crates/router/src/connector/globalpay.rs

```diff
@@ -442,8 +453,22 @@ impl ConnectorIntegration<api::Void, types::PaymentsCancelData, types::PaymentsR
         req: &types::PaymentsCancelRouterData,
         _connectors: &settings::Connectors,
     ) -> CustomResult<RequestContent, errors::ConnectorError> {
-        let connector_req = requests::GlobalpayCancelRequest::try_from(req)?;
-        Ok(RequestContent::Json(Box::new(connector_req)))
+        match req.request.minor_amount {
+            Some(amount) => {
+                let currency = req.request.currency.unwrap_or_default();
+                let amount =
+                    connector_utils::convert_amount(self.amount_converter, amount, currency)?;
+
+                let connector_router_data = requests::GlobalPayRouterData::from((amount, req));
+                let connector_req =
+                    requests::GlobalpayCancelRequest::try_from(&connector_router_data)?;
+                Ok(RequestContent::Json(Box::new(connector_req)))
+            }
+            None => {
+                let connector_req = requests::GlobalpayCancelRequest::try_from(req)?;
+                Ok(RequestContent::Json(Box::new(connector_req)))
+            }
+        }
     }
```

Comment:
what are we trying to achieve out here, instead can we do

```suggestion
        let amount =  req.request.minor_amount.and_then(| amount|{
            req.request.currency.map(|currency| {
                connector_utils::convert_amount(self.amount_converter, amount, currency)
            })
        }).transpose()?;
```